packages/store-indexer/src/postgres/createQueryAdapter.ts

```diff
@@ -14,70 +14,39 @@ import { decodeDynamicField } from "@latticexyz/protocol-parser";
-      const tables = (await getTables(database))
-        .filter((table) => address == null || getAddress(address) === getAddress(table.address))
-        .filter((table) => !tableIds.length || tableIds.includes(table.tableId));
+    async getLogs(opts) {
+      return getLogs(database, opts);
+    },
+    async findAll(opts) {
+      const filters = opts.filters ?? [];
+      const { blockNumber, logs } = await getLogs(database, {
+        ...opts,
+        // make sure we're always retrieving `store.Tables` table, so we can decode table values
+        filters: filters.length > 0 ? [...filters, { tableId: storeTables.Tables.tableId }] : [],
+      });
 
-      const tablesWithRecords = await Promise.all(
-        tables.map(async (table) => {
-          const sqlTable = buildTable(table);
-          const records = await database
-            .select()
-            .from(sqlTable)
-            .where(eq(sqlTable.__isDeleted, false))
-            .execute()
-            // Apparently sometimes we can have tables that exist in the internal table but no relation/schema set up, so queries fail.
-            // See https://github.com/latticexyz/mud/issues/1923
-            // TODO: make a more robust fix for this
-            .catch((error) => {
-              console.error(
-                "Could not query for records, returning empty set for table",
-                getTableName(sqlTable),
-                error
-              );
-              return [];
-            });
-          const filteredRecords = !filters.length
-            ? records
-            : records.filter((record) => {
-                const keyTuple = decodeDynamicField("bytes32[]", record.__key);
-                return filters.some(
-                  (filter) =>
-                    filter.tableId === table.tableId &&
-                    (filter.key0 == null || filter.key0 === keyTuple[0]) &&
-                    (filter.key1 == null || filter.key1 === keyTuple[1])
-                );
-              });
-          return {
-            ...table,
-            records: filteredRecords.map((record) => ({
-              key: Object.fromEntries(Object.entries(table.keySchema).map(([name]) => [name, record[name]])),
-              value: Object.fromEntries(Object.entries(table.valueSchema).map(([name]) => [name, record[name]])),
-            })),
-          };
-        })
-      );
+      const tables = logs.filter(isTableRegistrationLog).map(logToTable);
 
-      const internalTables = buildInternalTables();
-      const metadata = await database
-        .select()
-        .from(internalTables.chain)
-        .where(eq(internalTables.chain.chainId, chainId))
-        .execute();
-      const { lastUpdatedBlockNumber } = metadata[0] ?? {};
+      const tablesWithRecords: TableWithRecords[] = tables.map((table) => {
+        const records = logs
+          .filter((log) => getAddress(log.address) === getAddress(table.address) && log.args.tableId === table.tableId)
```

Comment:
This does less querying than we were doing before but for the same amount of data.

Before when records were distributed across many tables, we had to query for a list of tables then query each table individually for their records.

Now we query for all logs (passing the filters to the SQL query), then we just group them by table for the purposes of `findAll`.

This is mostly a backwards compat thing for `findAll` with the new table structure. The clients will start using the new `getLogs` endpoint in a PR coming soon.

Reply:
Ahh gotcha, this part is just grouping the logs by table, not really filtering the data. Even though this is mostly used for backwards compatibility, I wonder should we slightly change the logic here to iterate through the logs once and group them by table (ie have an object with `address/tableId` as key and array as value, iterate over the logs to push to the right array, and then do `Object.values`), instead of iterating through all logs `number of table` times?

Reply:
will adjust here but I'd prefer to save these kinds of things for an optimization pass where we can set up some measurement tools etc. (assuming we even want to keep supporting this endpoint)

although it will prob be more performant, I find your suggested approach a bit harder to read and maintain, so it'd be good to know what perf we're trading off in terms of readability vs speed/memory

Reply:
agree with no premature optimization and instead prioritizing benchmarks to find the actual bottlenecks!

Reply:
ah so the other reason I did this like this is because we need the table schema to decode the log, and didn't want to create another lookup mapping